stan.cabal

```diff
@@ -163,6 +163,9 @@ library
                      , trial ^>= 0.0.0.0
                      , trial-optparse-applicative ^>= 0.0.0.0
                      , trial-tomland ^>= 0.0.0.0
+  if impl(ghc >= 9.6)
```

Comment:
It's more flexible to introduce an automatic Cabal flag, `True` by default. When building with GHC < 9.6 (or when available versions of `filepath` are constrained by other components of build plan) Cabal solver will automatically flip it. Also easier to coerce Stack to build by simply flipping the flag in `stack.yaml`.

Reply:
I've updated the branch, Is the new version what you had in mind?